server/utils/crmDisplayHelper.ts

```diff
@@ -1,12 +1,75 @@
 import { CrmResponse } from '@eqApi'
-import { Section, ShowOrHideWhen } from '@crmDisplay'
+import { Section, ShowOrHideWhen, Subsection } from '@crmDisplay'
 import _ from 'lodash'
 
 export const includeSection = <T extends CrmResponse>(section: Section, crmResponse: T): boolean => {
   if (section.hideWhen && conditionsMet(section.hideWhen, crmResponse)) {
     return false
   }
-  return !section.showWhen || conditionsMet(section.showWhen, crmResponse)
+
+  const include = !section.showWhen || conditionsMet(section.showWhen, crmResponse)
+
+  if (include && isSectionEmpty(section, crmResponse)) {
+    return false
+  }
+
+  return include
+}
+
+// Utility to check if a section is empty
+const isSectionEmpty = <T extends CrmResponse>(section: Section, crmResponse: T): boolean => {
+  if (!section.subsections || section.subsections.length === 0) {
+    return false
+  }
+
+  return section.subsections.every(subsection => isSubsectionEmpty(subsection, crmResponse))
+}
+
+export const isSubsectionEmpty = <T extends CrmResponse>(subsection: Subsection, crmResponse: T): boolean => {
+  // Check if fields are empty
+  if (subsection.fields && subsection.fields.length > 0) {
+    const allFieldsEmpty = subsection.fields.every(field => {
+      if ('apiField' in field) {
+        const value = getApiFieldValue(crmResponse, field.apiField)
+        return !fieldHasValue(value)
+      }
+      return false
+    })
+
+    if (!allFieldsEmpty) {
+      return false
+    }
+  }
+
+  // Check custom display if present
+  if (subsection.customDisplay) {
+    const customDisplayValue = getApiFieldValue(crmResponse, subsection.customDisplay.apiField)
+    return !fieldHasValue(customDisplayValue)
+  }
+
+  // Default to empty if no fields or custom display
+  return true
+}
+
+// Utility to determine if a value is "empty"
+export const fieldHasValue = (value: unknown): boolean => {
```

Comment:
Only used in this file, so export can be removed
```suggestion
const fieldHasValue = (value: unknown): boolean => {
```